shopfloor/services/zone_picking.py

```diff
@@ -535,15 +544,22 @@ def _scan_source_location(
             )
         else:
             # if no move line, narrow the list of move lines on the scanned location
-            response = self._list_move_lines(location)
-            message = self.msg_store.location_empty(location)
+            response = self._list_move_lines(
+                location, no_package=scan_location_or_pack_first
+            )
+            if not scan_location_or_pack_first:
+                message = self.msg_store.location_empty(location)
+            else:
+                message = self.msg_store.location_empty_scan_package(location)
```

Comment:
Generally when we have a `else` block I prefer to invert the checks to avoid the negation here + a comment
```suggestion
            # Location is empty: invite the user to scan a package if the option is enabled
            if scan_location_or_pack_first:
                message = self.msg_store.location_empty_scan_package(location)
            else:
                message = self.msg_store.location_empty(location)
```

Reply:
Updated